app/api/openai/route.js

```diff
@@ -0,0 +1,32 @@
+import { OpenAIApi, Configuration } from "openai";
+import { NextResponse } from 'next/server';
+
+
+const configuration = new Configuration({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+const openai = new OpenAIApi(configuration);
+
+export async function POST(req) {
+  console.log(req, "req");
+  const { messages } = await req.json()
+  console.log(messages, "messages");
+  if (messages !== undefined) {
+    const content = `If the following statement is a fact, meaning the result is either true or false, you should respond with one of the following: "Yes, you are right." or "No, you are not right this time." If it is not a fact and it's a question, please respond with: "Sorry, but you should ask a question that can be answered with either true or false". Try to respond to the following sentence following the rules we defined: ${messages}`;
```

Comment:
Instead of sending the instructions in each message, you may separate it to one message with a "system" role and one with a "user" role like the example in https://platform.openai.com/docs/guides/gpt/chat-completions-api